group-coordinator/src/test/java/org/apache/kafka/coordinator/group/assignor/RangeAssignorTest.java

```diff
@@ -730,7 +922,20 @@ private void assertAssignment(
         assertEquals(expectedAssignment.size(), computedGroupAssignment.members().size());
         for (String memberId : computedGroupAssignment.members().keySet()) {
             Map<Uuid, Set<Integer>> computedAssignmentForMember = computedGroupAssignment.members().get(memberId).partitions();
-            assertEquals(expectedAssignment.get(memberId), computedAssignmentForMember);
+            Map<Uuid, Set<Integer>> expectedAssignmentForMember = expectedAssignment.get(memberId);
+
+            // Compare the content of the maps by entries.
+            assertEquals(expectedAssignmentForMember.size(), computedAssignmentForMember.size());
+            for (Map.Entry<Uuid, Set<Integer>> entry : expectedAssignmentForMember.entrySet()) {
+                Set<Integer> expectedSet = entry.getValue();
+                Set<Integer> computedSet = computedAssignmentForMember.get(entry.getKey());
+
+                // Convert both sets to HashSet for comparison.
+                Set<Integer> normalizedExpectedSet = new HashSet<>(expectedSet);
+                Set<Integer> normalizedComputedSet = new HashSet<>(computedSet);
+
+                assertEquals(normalizedExpectedSet, normalizedComputedSet);
+            }
```

Comment:
Do we really need this? I was expecting the comparison to work with RangeSet.

Reply:
yeah, I think it didn't work initially, but changed it now